libs/shared/storage/src/lib/shared-storage.service.ts

```diff
@@ -153,28 +159,28 @@ export class SharedStorageService {
     } catch (err) {
       if (err === timeoutErr) {
         this.logger.error(
-          `Timeout when uploading '${id}' to storage in ${this.S3_UPLOAD_TIMEOUT_TIME} ms.`,
+          `Timeout when uploading '${key}' to storage in ${this.S3_UPLOAD_TIMEOUT_TIME} ms.`,
         );
-        throw new Error('Timeout when uploading file to storage.');
+        throw new RequestTimeoutException('Timeout when uploading file to storage.');
       } else {
         const details =
           err instanceof Error && err.message
             ? err?.message
-            : `Error when uploading ${id} to storage.`;
+            : `Error when uploading ${key} to storage.`;
         this.logger.error(details);
 
         const message =
           err instanceof Error && err.message
             ? err?.message
             : 'Error when uploading file to storage.';
-        throw new Error(message);
+        throw new InternalServerErrorException(message);
```

Comment:
I don't think we should use http exceptions outside of the controller/api and the handlers are designed to turn these into specific responses. For errors that we catch, we can use BiosimulationsError so that we have better control of the fields and can make better error messages

Reply:
I tried to make the error more informative than just generic `Error`.

Scopes prevent `BiosimulationsError` from being imported here.

Reply:
`BiosimulationsException` should be used here. I changed errors to `BiosimulationsException`.